### DISCUSSION TITLE
Can i do file compression / archive / zipping ?

### DISCUSSION THREAD

**Discussion by sachinsenal0x64** [sxyazi/yazi#538] [1. Q&A] [answered]
Hey It's better to have file compressions like gz also file zipping

**Comment by Reledia** [comment]
If you don't want to use the shell inside ya, I guess you could define execs setting your preferred options for compressing and uncompressing, and associate them with the right mime types (like `application/zip`)

**Comment by sxyazi** [comment, marked as answer]
See https://github.com/sxyazi/yazi/issues/51#issuecomment-1763697521

**Reply by sabisho** [reply]
I think this should be a core feature instead of plugin.
Still you are the Developer so you know what's best 😊

**Reply by sxyazi** [reply]
It's not a plugin, just call `zip` through Yazi's [`shell` command](https://yazi-rs.github.io/docs/configuration/keymap/#shell) to achieve compression.

Do you mean to make Yazi built-in `zip`?

**Reply by sabisho** [reply]
> Do you mean to make Yazi built-in zip?

Yes

**Comment by sachinsenal0x64** [comment]
> See [#51 (comment)](https://github.com/sxyazi/yazi/issues/51#issuecomment-1763697521)

Thanks, anyway, why preview (image) not work when I use zellij ?

**Reply by sxyazi** [reply]
See https://github.com/sxyazi/yazi/issues/334#issuecomment-1793080857

**Reply by sachinsenal0x64** [reply]
Is there a way to integrate a custom preview module into Yazi ? I am currently testing a module that can display images inside zellij, and I'm looking for guidance on integrating it seamlessly with Yazi.

![image](https://github.com/sxyazi/yazi/assets/127573781/601f6211-9c55-4413-b765-8453159eb78c)
